pctiler/pctiler/endpoints/pg_mosaic.py

```diff
@@ -1,84 +1,23 @@
-from typing import Any, Callable, Dict
+from dataclasses import dataclass
 
-from fastapi import Depends, Query
-from starlette.requests import Request
-from titiler.pgstac.factory import MosaicTilerFactory as PgstacMosaicTilerFactory
-from titiler.pgstac.models import SearchQuery
+from fastapi import Query
+from titiler.core import dependencies
+from titiler.pgstac.factory import MosaicTilerFactory
 
 from pctiler.colormaps import PCColorMapParams
 from pctiler.config import get_settings
-from pctiler.db import Connection, with_retry_connection
-from pctiler.reader import CustomPGSTACBackend
+from pctiler.reader import PGSTACBackend
 
 
-def AdditionalTileParam(
-    collection: str = Query(None, description="JSON encoded custom Colormap"),
-) -> Dict[str, Any]:
-    return {"collection": collection}
+@dataclass
+class AssetsBidxExprParams(dependencies.AssetsBidxExprParams):
 
+    collection: str = Query(None, description="STAC Collection ID")
```

Comment:
we add `collection` to `AssetsBidxExprParams` params which will be used in all endpoints